spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,137 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+  describe 'GET users#index' do
+
+    before do
+      create(:user)
+      create(:user)
+    end
+
+    it 'returns succssfully' do
+      get :index
+      expect(response).to be_successful
+    end
+
+    it 'follows the routes successfully' do
+      get :index
+      expect(controller.controller_path).to eq('users')
+    end
+
+    it 'renders the index template' do
+      get :index
+      expect(response).to render_template("index")
+    end
+
+    it 'compares that all users in the table are equal to the index instance variable' do
+      get :index
+      expect(User.all).to match(assigns(:users))
+    end
+  end
+
+  describe 'GET#new' do
+    it 'returns succssfully' do
+      get :new
+      expect(response).to be_successful
+    end
+
+    it 'proves that @user was instantiated correctly' do
+      get :new
+      expect(controller.instance_variable_get(:@user)).to be_a_new(User)
+    end
+  end
+
+  describe 'POST users#create' do
+
+    context 'if user is valid' do
+
+      before do
+        post :create, params: { user: FactoryBot.attributes_for(:user) }
+      end
+
+      it 'assigns a newly created user as @user' do
+        expect(controller.instance_variable_get(:@user)).to be_a(User)
+      end
+
+      it 'redirects to root path' do
+        expect(response).to redirect_to(root_path)
+      end
+
+      it 'redirect 302' do
+        expect(response).to have_http_status(302)
+      end
+    end
+
+    context 'if user is invalid' do
+
+      before do
+        post :create, params: { user: { first_name: 'j', last_name: 's'} }
+      end
+
+      it 'renders new' do
+        expect(subject).to render_template(:new)
+      end
+
+      it 'does not redirect to root_path' do
+        expect(subject).to_not redirect_to(root_path)
+      end
+    end
+  end
+
+  describe 'GET #show' do
+
+    before do
+      create(:user)
+    end
+
+    it 'returns succssfully' do
+      get :show, params: { id: User.first.id }
+      expect(response).to be_successful
+    end
+
+    it 'renders show template' do
+      get :show, params: { id: User.first.id }
+      expect(response).to render_template(:show)
+    end
+
+    it 'assigns @user correctly' do
+      get :show, params: { id: User.first.id }
+      expect(controller.instance_variable_get(:@user)).to eql(User.first)
+    end
+  end
+
+  describe 'DELETE #destroy' do
+
+    before do
+      create(:user)
```

Comment:
You can use the times loop or use factory bot create_list

Reply:
Done!